src/compiler/config/validate-config.ts

```diff
@@ -104,24 +104,43 @@ export const validateConfig = (
   }
 
   validatedConfig.extras = validatedConfig.extras || {};
-  validatedConfig.extras.appendChildSlotFix = !!validatedConfig.extras.appendChildSlotFix;
-  validatedConfig.extras.cloneNodeFix = !!validatedConfig.extras.cloneNodeFix;
   validatedConfig.extras.lifecycleDOMEvents = !!validatedConfig.extras.lifecycleDOMEvents;
   validatedConfig.extras.scriptDataOpts = !!validatedConfig.extras.scriptDataOpts;
-  validatedConfig.extras.slotChildNodesFix = !!validatedConfig.extras.slotChildNodesFix;
   validatedConfig.extras.initializeNextTick = !!validatedConfig.extras.initializeNextTick;
   validatedConfig.extras.tagNameTransform = !!validatedConfig.extras.tagNameTransform;
-  // TODO(STENCIL-914): remove `experimentalSlotFixes` when it's the default behavior
-  validatedConfig.extras.experimentalSlotFixes = !!validatedConfig.extras.experimentalSlotFixes;
 
-  // if the user has set `extras.experimentalSlotFixes` then we turn on all of
-  // the slot-related fixes automatically.
+  // TODO(STENCIL-914): remove when `experimentalSlotFixes` is the default behavior
+  // If the user set `experimentalSlotFixes` and any individual slot fix flags to `false`, we need to log a warning
+  // to the user that we will "override" the individual flags
+  if (validatedConfig.extras.experimentalSlotFixes === true) {
+    const possibleFlags: (keyof ValidatedConfig['extras'])[] = [
```

Comment:
I think this array is causing a couple new SNC violations.

Because `ValidatedConfig#Extras` is an optional type, accessing it as `ValidatedConfig['extras']` creates a type of `never` (since it can be `undefined`).

I think this should be fixable by using the `ConfigExtras` type directly:
```suggestion
    const possibleFlags: (keyof ConfigExtras)[] = [
```